### PR TITLE
Landscape support

### PR DESCRIPTION
**Pull request by aliseyalvi**
### What does this PR?
This PR enables landscape mode in Ecency.  
It enables the landscape support in the whole app and uses react-native-orientation-locker package to lock the orientation to portrait in application screen. Orientation is only unlocked in post screen and locked again on coming back. From this point we can add support for landscape in each screen and unlock the orientation lock in that screen.

### Steps to reproduce
On feed screen enable rotation and turn screen to landscape. App would not crash as it crashes previously. But it would remain in portrait due to orientation lock. 
Now tap on any post and go to post screen. Turn screen to landscape again and you will see app in landscape mode with some distorted layout. 

### Issue number
#2093. #2147 . #2128. 
### Screenshots/Video

Screenshot_20220326-181335
Screenshot_20220326-181406

**UPDATE**
landscape content is now centred and have width equivalent of portrait width to avoid re-render.
Screenshot 2022-04-06 at 11 50 54 PM

### CODE REVIEW

**Comment by aliseyalvi** [comment]
@noumantahir please check the latest changes.

- center aligned the post body when landscape is active.
- changed sticky tab bar height and padding

https://user-images.githubusercontent.com/48380998/161905859-a6458dd2-2166-4957-afae-486071abf63c.mp4

File: src/components/postView/view/postDisplayView.js [hunk 1]
```diff
@@ -20,8 +22,10 @@ import { ParentPost } from '../../parentPost';
 // Styles
 import styles from './postDisplayStyles';
 import { OptionsModal } from '../../atoms';
+import { orientations } from '../../../redux/constants/orientationsConstants';
 
 const HEIGHT = Dimensions.get('window').width;
```
Comment (noumantahir) [inline src/components/postView/view/postDisplayView.js:27]: @aliseyalvi height is equal to width here, could be a typo

File: src/components/postView/view/postDisplayView.js [hunk 2]
```diff
@@ -50,8 +56,21 @@ const PostDisplayView = ({
...
+  const [contentWidth, setContentWidth] = useState(WIDTH - 32);
+  const deviceOrientation = useSelector((state) => state.ui.deviceOrientation);
 
   // Component Life Cycles
+  useEffect(() => {
+    if (
+      deviceOrientation === orientations.LANDSCAPE_LEFT ||
+      deviceOrientation === orientations.LANDSCAPE_RIGHT
+    ) {
+      setContentWidth(Dimensions.get('window').height - 128);
```
Comment (noumantahir) [inline src/components/postView/view/postDisplayView.js:68]: @aliseyalvi shouldn't this change since now we do not want to change contentWidth

**Review by noumantahir — CHANGES_REQUESTED** [review]
Added code comments, and this UI glitch needs testing.

Overall, this seems to have fixed the packager reset problem, tested on a 2014 phone emulator, do not have any that much old real device to test

Screenshot 2022-04-06 at 11 06 51 PM

File: src/components/postView/view/postDisplayView.js (hunk 2 above)
Comment (aliseyalvi) [reply src/components/postView/view/postDisplayView.js:68]: yes, will change that

File: src/components/postView/view/postDisplayView.js (hunk 1 above)
Comment (aliseyalvi) [reply src/components/postView/view/postDisplayView.js:27]: it coule de a typo

**Review by noumantahir — APPROVED** [review]

**Comment by noumantahir** [comment]
Everything seems on point from my side, @good-karma looking forward to your feedback....

**Review by feruzm — APPROVED** [review]
Lgtm, need some more testing on different devices

**Pull request merged**